### DISCUSSION TITLE
different facet counts between algoliasearch-helper 3.22.5 and 3.22.6

### DISCUSSION THREAD

**Discussion by stefanprobst** [algolia/instantsearch#6482] [Q&A] [answered]
a recent change between `algoliasearch-helper` 3.22.5 and 3.22.6 has changed the facet counts - i assume the cause is https://github.com/algolia/instantsearch/pull/6445

with 3.22.5 i see:

![algoliasearch-helper-3 22 5](https://github.com/user-attachments/assets/3c01a298-b672-4c62-86b5-9edc2502fa44)

with 3.22.6 i see:

![algoliasearch-helper-3 22 6](https://github.com/user-attachments/assets/54e46498-cdf1-4899-87a7-623b9cdacbcd)

AFAICT the counts in 3.22.5 are taken from the extra facet query (where no facet filter is applied), i.e. they display how many documents have e.g. "Kämmerer" in the list of values.

in 3.22.6 however, the counts are taken from the main query, i.e. they represent not how many documents have "Kämmerer", but how many documents have "Kämmerer" AND any currently selected/refined values - and these values differ rather substantially (1702 vs. 174).

can you elaborate why this was changed? thanks!

**Comment by Haroenv** [comment]
You're right that this change will be related to the change made in #6445, and was intentional. I'll now look into where this discrepancy is coming from in your case. Are both queries exhaustive? It's odd that all numbers are the same except those two.

I wonder if the logic should be tweaked to prefer the facets which are set as exhaustive if only one of the two have that; or otherwise taking the largest number.

Is this reproducible with all facets or just one? Do you have a reproduction (can be a curl or just the `facets` and `exhaustive` result) available?

**Reply by stefanprobst** [reply]
sorry for the late reply.

you can find a repro here: https://github.com/stefanprobst/issue-algolia-instantsearch/

see the [readme](https://github.com/stefanprobst/issue-algolia-instantsearch/blob/main/README.md) for the two issues documented there (the first one is the issue from this discussion, the second issue is about the fact that selected facet values are not guaranteed to be included in the response, which only has the top maxFacetValues values, so their count will be incorrectly displayed as 0) 

to see the issue from this discussion, go to: http://localhost:3000/?persons%5BrefinementList%5D%5Bfunctions%5D%5B0%5D=Geh.%20Rat%2C%20wirkl.

**Reply by stefanprobst** [reply]
@Haroenv just making sure you have seen the above message with a reproduction. thanks

**Reply by Haroenv** [reply]
Oh, I had not seen that, sorry.

**Reply by Haroenv** [reply, marked as answer]
https://github.com/algolia/instantsearch/pull/6536 should fix your use case. Thanks for bringing it up. 

Your second question I can't really help with, that's an engine feature request as we can't force values to be in the response. A workaround can be to put an artificially high `maxValuesPerFacet` in a configure widget. This will make your response slightly larger, but makes it more likely for the facet to be present in the response.

**Reply by Haroenv** [reply]
Have you tried out the preview @stefanprobst? feel free to comment on the PR to affirm our assumptions are correct :)
